Codecademy/gamut
# feat(DropdownButton): Auto position for dropdown button popover

### Overview



Use 'above' position for popover with dropdown button when close to viewport bottom



### PR Checklist

- [x] Related to JIRA ticket: [REACH-2522]

#### Testing Instructions

- View the button dropdown on the preview env [here](https://643054042836db7811b7b4eb--gamut-preview.netlify.app/?path=/docs/brand-labs-molecules-dropdownbutton--dropdown-button).
- Make your browser window short enough that the `click me` button is near the bottom of the window.
- Click it and verify that the popover renders above the button rather than below.
- Scroll while the popover is open to verify that it only renders above when there is not enough room for it to render below.

#### PR Links and Envs

| Repository   | PR Link                                                  |
| :----------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/35449)  |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/1714)  |




[REACH-2522]: https://codecademy.atlassian.net/browse/REACH-2522?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
